web/pdf_document_properties.js

```diff
@@ -87,71 +113,71 @@ class PDFDocumentProperties {
    * @param {Object} pdfDocument - A reference to the PDF document.
    * @param {string} url - The URL of the document.
    */
-  setDocumentAndUrl(pdfDocument, url) {
+  setDocument(pdfDocument, url) {
+    if (this.pdfDocument) {
```

Comment:
I think there is no real need to check for this here since `this._reset()` does the check too.

Moreover, shouldn't `this.pdfDocument` be private as well, i.e., `this._pdfDocument`, as it may only be set using this setter (like `this._url`)?